src/main/kotlin/uk/gov/gdx/datashare/services/OutboundEventQueueService.kt

```diff
@@ -302,4 +321,50 @@ class OutboundEventQueueService(
       AwsQueue(it, sqsClient, it)
     }
   }
+
+  private fun getQueueMetrics(queueNames: List<String>): Map<String, QueueMetric> {
+    val dlqLengthQueries = queueNames.map {
+      buildMetricDataQuery("ApproximateNumberOfMessagesVisible", dlqName(it))
+    }
+    val ageOfOldestMessageQueries = queueNames.map {
+      buildMetricDataQuery("ApproximateAgeOfOldestMessage", it)
+    }
+
+    val getMetricDataRequest = GetMetricDataRequest.builder()
+      .metricDataQueries(dlqLengthQueries + ageOfOldestMessageQueries)
+      .startTime(Instant.now().minusSeconds(120))
+      .endTime(Instant.now())
+      .build()
+
+    val metricDataResults = cloudWatchClient.getMetricData(getMetricDataRequest).metricDataResults()
+    return queueNames.associateWith { queueName ->
+      QueueMetric(
+        metricDataResults.find { it.id() == queueName }?.values()?.firstOrNull()?.toInt() ?: 0,
+        metricDataResults.find { it.id() == dlqName(queueName) }?.values()?.firstOrNull()?.toInt() ?: 0,
```

Comment:
Do we want these to be zero on missing or something else? Surely missing is quite concerning and we want to be able to alarm on it?